src/vm/gchelpers.cpp

```diff
@@ -60,6 +60,45 @@ inline gc_alloc_context* GetThreadAllocContext()
     return & GetThread()->m_alloc_context;
 }
 
+// Checks to see if the given allocation size exceeds the
+// largest object size allowed - if it does, it throws
+// an OutOfMemoryException with a message indicating that
+// the OOM was not from memory pressure but from an object
+// being too large.
+inline void CheckObjectSize(size_t alloc_size)
+{
+    CONTRACTL {
+        THROWS;
+        GC_TRIGGERS;
+    } CONTRACTL_END;
+
+    size_t min_obj_size = sizeof(uint8_t*)   // sync block
```

Comment:
> first of all, this calculation is incorrect, min_obj_size is 3-ptr size, not 4;

You're right, the error here is in the alignment - I misread the `ALIGNCONST` macro, I apologize.

How would you like this to be exposed? `LARGE_OBJECT_SIZE` is already `#defined` in `gcinterface.h`, so would you be okay with defining `min_obj_size` there?